yogstation/code/game/gamemodes/gangs/dominator.dm

```diff
@@ -0,0 +1,242 @@
+#define DOM_BLOCKED_SPAM_CAP 6
+#define DOM_REQUIRED_TURFS 30
+#define DOM_HULK_HITS_REQUIRED 10
+
+/obj/machinery/dominator
+	name = "dominator"
+	desc = "A visibly sinister device. Looks like you can break it if you hit it enough."
+	icon = 'icons/obj/machines/dominator.dmi'
+	icon_state = "dominator"
+	density = TRUE
+	anchored = TRUE
+	layer = HIGH_OBJ_LAYER
+	max_integrity = 300
+	integrity_failure = 100
+	move_resist = INFINITY
+	armor = list("melee" = 20, "bullet" = 50, "laser" = 50, "energy" = 50, "bomb" = 10, "bio" = 100, "rad" = 100, "fire" = 10, "acid" = 70)
+	var/datum/team/gang/gang
+	var/operating = FALSE	//false=standby or broken, true=takeover
+	var/warned = FALSE	//if this device has set off the warning at <3 minutes yet
+	var/spam_prevention = DOM_BLOCKED_SPAM_CAP //first message is immediate
+	var/datum/effect_system/spark_spread/spark_system
+	var/obj/effect/countdown/dominator/countdown
+
+/obj/machinery/dominator/Initialize()
+	set_light(2)
+	GLOB.poi_list |= src
+	spark_system = new
+	spark_system.set_up(5, TRUE, src)
+	countdown = new(src)
+	update_icon()
+	.=..()
+
+/obj/machinery/dominator/Destroy()
+	if(!(stat & BROKEN))
+		set_broken()
+	GLOB.poi_list.Remove(src)
+	gang = null
+	QDEL_NULL(spark_system)
+	QDEL_NULL(countdown)
+	STOP_PROCESSING(SSmachines, src)
+	return ..()
+
+/obj/machinery/dominator/emp_act(severity)
+	take_damage(100, BURN, "energy", 0)
```

Comment:
I would recommend making it just turn off temporarily, because otherwise chemists can kill it so easily.

Reply:
Prob just going to nerf the damage emps do so you would have to emp it like 5 times to destroy it